datafusion/functions-array/src/kernels.rs

```diff
@@ -306,9 +306,15 @@ pub(super) fn gen_range(args: &[ArrayRef], include_upper: bool) -> Result<ArrayR
             );
         }
         // Below, we utilize `usize` to represent steps.
-        // But on 32-bit targets, the absolute value of `i64` may fail to fit into `usize`.
+        #[cfg(target_pointer_width = "64")]
+        let step_abs = step.unsigned_abs() as usize;
+        // On 32-bit targets, the absolute value of `i64` may fail to fit into `usize`.
+        #[cfg(not(target_pointer_width = "64"))]
         let step_abs = usize::try_from(step.unsigned_abs()).map_err(|_| {
```

Comment:
If `usize::try_from(step.unsigned_abs()).map_err` can handle whatever the target size is, I think we don't need to split the code for two condition

Reply:
If we are running in 32 bit target machine, do we able to reach the code here and get step with i64? 🤔

Reply:
> If `usize::try_from(step.unsigned_abs()).map_err` can handle whatever the target size is, I think we don't need to split the code for two condition

Perhaps simpler is better. Let me merge them.

Reply:
> If we are running in 32-bit target machine, are we able to reach the code here and get the `step` with i64? 🤔

I think it's possible, because `step` comes from user input.